### PR TITLE
proxmox_kvm: light integration of proxmox module_utils

### PR DESCRIPTION
**Pull request by tleguern**
##### SUMMARY

Use the common authentication code without restructuring the whole code.
While there ~~drop precautions for versions before Proxmox VE 4.0, everyone had six years to upgrade.~~ put parameter checks at the right place.

Should not change existing behaviours.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

proxmox_kvm

### CODE REVIEW

**Comment by Thulium-Drake** [comment]
\o/ for cleaning up old warnings :-)

Looks good! Thanks for this!

**Comment by felixfontein** [comment]
> While there drop precautions for versions before Proxmox VE 4.0, everyone had six years to upgrade.

Be warned that this makes it impossible to merge this before 3.0.0. Breaking changes are only allowed in major releases.

**Comment by Aversiste** [comment]
> Be warned that this makes it impossible to merge this before 3.0.0. Breaking changes are only allowed in major releases.

I did not thought of that. I can always add them back until 3.0.0 (sorry @Thulium-Drake !).

File: plugins/modules/cloud/misc/proxmox_kvm.py [hunk 1]
```diff
@@ -1137,20 +1132,21 @@ def main():
...
+        if module.params['force']:
+            module.fail_json(msg="This feature only works with Proxmox VE version 4 and onwards: force")
+        for p in ['protection', 'skiplock']:
+            if module.params[p] is not None:
+                module.fail_json(msg="This feature only works with Proxmox VE version 4 and onwards: {0}".format(p))
+    # These features only work since Proxmox VE 6.0
+    if pve_version < LooseVersion('6.0'):
+        for p in ['ciuser', 'cipassword', 'sshkeys', 'ipconfig']:
+            if module.params[p] is not None:
+                module.fail_json(msg="This feature only works with Proxmox VE version 6 and onwards: {0}".format(p))
```
Comment (Aversiste) [inline plugins/modules/cloud/misc/proxmox_kvm.py:1149]: As discussed on IRC chatroom #ansible-community call to fail_json should be replaced with warnings and the parameters be nullified.

**Comment by felixfontein** [comment]
@tleguern ping to check whether you are still working on this :)

**Comment by tleguern** [comment]
> @tleguern ping to check whether you are still working on this :)

I did not make any progress at all on this PR so far but I am willing to.

**Comment by felixfontein** [comment]
@tleguern regular reminder ping :)

**Comment by felixfontein** [comment]
@tleguern please rebase this, otherwise I think it's best to close this.

needs_info

**Comment by tleguern** [comment]
Hello,

Sorry I do not have time allocated to proxmox for now. It's best to close it and I will pick it up later.

**Pull request closed without merging**